libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetwork/Classes/Network/Queue/CSFNetwork.m

```diff
@@ -304,11 +302,11 @@ - (void)URLSession:(NSURLSession *)session downloadTask:(NSURLSessionDownloadTas
 - (void)receivedDevicedUnauthorizedError:(CSFAction *)action {
 }
 
-#pragma mark - Handling Login State
+#pragma mark - SFAuthenticationManagerDelegate
 
-- (void)userWillLogOutNotification:(NSNotification *)notification {
-    SFUserAccount *userAccount = notification.userInfo[kSFUserAccountKey];
-    [[self class] cleanupNetworkForUserAccount:userAccount];
+- (void)authManager:(SFAuthenticationManager *)manager willLogoutUser:(SFUserAccount *)user {
+    [[self class] removeSharedInstance:user];
```

Comment:
I dunno, I fixed a similar bug recently where where a class's delegate nil'ed out the class instance as part of its delegate method implementation, and subsequent lines in the class made the app crash, because the memory had been deallocated.  So it's certainly something that *can* come up.  Interestingly, that crash only happened on device, not the simulator.

Anyway, I just wanted to point it out.  It's probably fine.